### PR TITLE
Fix several mismatches to build docker container and library itself.

### PR DESCRIPTION
**Pull request by lazykyama**
This PR should be related to three issues.

* #193
    - This is not directly related to main issue but might cause sometimes a few errors on user codes using `onnx-tensorrt`.
* #246
* #248 

BTW, I explicitly added a path to CUDA header files directory into `CMakeLists.txt` but it might not be a best solution... To keep `CMakeLists.txt` clean, adding such information into only `Dockerfile` might be better. What do you think, team? If the latter solution is better, please let me know. I'll fix my PR.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +174,8 @@ endif()
 # Specify the cuda host compiler to use the same compiler as cmake.
 set(CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
 
+# Add an include headers' directory for CUDA, explicitly.
+set(CUDA_INCLUDE_DIRS /usr/local/cuda/include/)
```
Comment (leimao) [inline CMakeLists.txt]: We would better not to hard code this. Instead, using 
```
cmake -DCUDA_INCLUDE_DIRS=/usr/local/cuda/include/ ..
```
might be better.

**Comment by kevinch-nv** [comment]
Closing this in favor of #256

**Comment by lazykyama** [comment]
@kevinch-nv 
I'd like to merge the following commit because this causes issue #193. However, PR #256 doesn't include this commit. Is it ok that I make a new PR on this?
6e6b547ed21fc1742afc70c6cfcb5eece60d8be7

**Comment by kevinch-nv** [comment]
@lazykyama Yes, that would be helpful. Thanks!

**Comment by lazykyama** [comment]
Ok. I will make new PR soon!

**Pull request closed without merging**